framework/src/base/BlockRestrictable.C

```diff
@@ -201,6 +221,8 @@ BlockRestrictable::hasBlocks(std::vector<SubdomainID> ids) const
 bool
 BlockRestrictable::hasBlocks(std::set<SubdomainID> ids) const
 {
+  mooseAssert(_initialized, "BlockRestrictable not initialized");
```

Comment:
Nah, There's little point in doing that. You'd have to manufacture a broken
object. There's nothing wrong with not testing asserts. They are there
because we expect them to never be hit, but they are good indicators when
things do go wrong and your assumptions break down.

On Thu, May 26, 2016 at 2:27 PM Andrew E Slaughter <notifications@github.com>
wrote:

> In framework/src/base/BlockRestrictable.C
> <https://github.com/idaholab/moose/pull/7089#discussion_r64815027>:
>
> > @@ -201,6 +221,8 @@ BlockRestrictable::hasBlocks(std::vector<SubdomainID> ids) const
> >  bool
> >  BlockRestrictable::hasBlocks(std::set<SubdomainID> ids) const
> >  {
> > +  mooseAssert(_initialized, "BlockRestrictable not initialized");
>
> Should we add a test that hits one of these asserts?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/idaholab/moose/pull/7089/files/d99e945961c595f44e89f98e15de5d0c2c23ef19#r64815027>
>